js_modules/dagit/packages/core/src/launchpad/LaunchpadSingleSessionContainer.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+
+import {
+  createSingleSession,
+  IExecutionSessionChanges,
+  useInitialDataForMode,
+} from '../app/ExecutionSessionStorage';
+import {RepoAddress} from '../workspace/types';
+
+import LaunchpadSessionContainer from './LaunchpadSessionContainer';
+import {LaunchpadSessionContainerPartitionSetsFragment} from './types/LaunchpadSessionContainerPartitionSetsFragment';
+import {LaunchpadSessionContainerPipelineFragment} from './types/LaunchpadSessionContainerPipelineFragment';
+
+interface Props {
+  pipeline: LaunchpadSessionContainerPipelineFragment;
+  partitionSets: LaunchpadSessionContainerPartitionSetsFragment;
+  repoAddress: RepoAddress;
+}
+
+export const LaunchpadSingleSessionContainer = (props: Props) => {
+  const {pipeline, partitionSets, repoAddress} = props;
+
+  const initialDataForMode = useInitialDataForMode(pipeline, partitionSets);
+
+  const [session, saveSession] = React.useState(() => createSingleSession(initialDataForMode));
+  const onSaveSession = React.useCallback((changes: IExecutionSessionChanges) => {
+    saveSession((current) => ({...current, ...changes}));
```

Comment:
I'm not sure it matters, but the applyChangesToSession helper had this line in it that isn't in the code path anymore:

```
  if (changes.runConfigYaml && changes.runConfigYaml !== saved.runConfigYaml && saved.runId) {
    changes.configChangedSinceRun = true;
  }
```

Maybe that could be pushed down into the call to onSave within the LaunchpadSessionContainer?

EDIT: I actually don't see anywhere `configChangedSinceRun` is read... maybe that property and that logic should actually get deleted.

Reply:
Yeah, maybe it's become obsolete. I think soon if we're launching from a run config, we're going to do it from a dialog launchpad anyway, so I think this won't matter at that point either way.